### ISSUE TITLE
small improvement to DAGCircuit.replace_block_with_op()

### ISSUE THREAD

**Issue by aeddins-ibm** [Qiskit/qiskit-terra#8724]
### What should we add?

Short version: 
- Add the line `return new_node` to the end of `DAGCircuit.replace(block_with_op)`.

Details:
`DAGCircuit.replace_block_with_op(block, op, ...)` internally creates a `new_node` containing `op`, and replaces the `block` of DAG nodes by this `new_node`.

I think `replace_block_with_op()` should return `new_node`. (Currently there is no `return` statement). This change would be in line with the behavior of `DAGCircuit.substitute_node()`, which does return its internal `new_node`. Otherwise, the user does not know the node_id or even the name of the new node, so finding the new node for further manipulations is difficult. (All I can think of is to look along one of the wires for the one node in the new DAG that wasn't in the old DAG... obviously this is cumbersome.) Since the function already has `new_node` available internally, seems like we may as well `return` it.

**Comment by jakelishman** [comment]
This sounds very sensible to me. Would you be able to make the PR for it?

**Comment by enavarro51** [comment]
@aeddins-ibm and @jakelishman. I can take care of this if you like.

**Comment by jakelishman** [comment]
Yes please, Edwin - I'm guessing Andrew doesn't have time to do it himself.  I'll assign you - there's no particular rush because we won't merge any PR before Terra 0.22.0rc1 goes out (hopefully Thursday, but both Matthew and I are quite busy on other things as well).

**Comment by aeddins-ibm** [comment]
Thanks @enavarro51 , that would be great! (Sorry @jakelishman , I kept intending to come back to this but never found the time)

**Comment by jakelishman** [comment]
No worries at all - there's no rush from our side!
